gcc/rust/typecheck/rust-hir-type-check-expr.cc

```diff
@@ -1289,8 +1289,8 @@ TypeCheckExpr::visit (HIR::BreakExpr &expr)
       TyTy::BaseType *loop_context = context->peek_loop_context ();
       if (loop_context->get_kind () == TyTy::TypeKind::ERROR)
 	{
-	  rust_error_at (expr.get_locus (),
-			 "can only break with a value inside %<loop%>");
+	  rust_error_at (expr.get_locus (), ErrorCode ("E0571"),
+			 "%<break%> with value from a %<while%> loop");
```

Comment:
- Changed the message similiar to rustc - [`E0571`](https://doc.rust-lang.org/error_codes/E0571.html)
- This code only works for `while` loop.
- Can you tell me how should the get the name of the loop we are using. So, we can make this error more generic..

Reply:
> Can you tell me how should I get the name of the loop we are using ? So, we can make this error more generic..

I don't think there is a way at the moment to get that information. So we should keep the original message, which was `can only break with a value inside a %<loop%> block`

> This code only works for while loop.

What do you mean?

Reply:
> I don't think there is a way at the moment to get that information. So we should keep the original message, which was `can only break with a value inside a %<loop%> block`


Sure..
 

> What do you mean?

Oh, I was unable to think for an example to test gccrs on `for` loop, for this example ([`godbolt`](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DEArgoKkl9ZATwDKjdAGFUtEywYgA7KUcAZPAZMADl3ACNMYhAAZlIAB1QFQjsGFzcPb3jE5IEAoNCWCKjYq0wbFKECJmICNPdPH1LygUrqgjyQ8MiYyyqauozGvvbAzsLu6IBKS1QTYmR2DioGAGoWJkCISZWAUi8AIR2NAEEVs5X6AjWTK7xd6IARFa4d6MOT85WAei/L66vqsR7k8dgBWfZcUgAJlIsWkoNIADZSD4ABykACcpC4GmxkK4MK4sSk2IRXGRXB8XHRXCxUNxUMhUJhUNiUOkUIRUORUJ8UPRUKx0Q0YIer3ep3OyxWCiYtgUNEw6AgnhWZjwAC9MNsALSvbArMKoVy7A4rVY7TkrVnA15PDSmsUfc5/YiYBQmWhXO0rfhAu6BZ4AOiDoIdewln0%2BeCoMrleAVeCVEDw2wjR0lUajYTdTAA1taVgAqFZ4cUZrNnPZOzOO8snasZjjTWicUG8TwcLSkVCcABKZiuClm80wu1ZPFIBE0zemeZAvKDki8iNRoNRBNRqI0GmivP0nEkHZnPc4vAUIFx067zdIcFgMEQKFQLDidEi5EoaFf76iwGQyDEAoOoKMwcQKAgqAWFgABueALAAakmADuADycSMJwk40F6kQXhAYQnmEgTVAAnlhvDEcwxCkahYTaGU16Tt%2BbCCKhDC0ORN6kFgYQmMAThiLQF7cLwWDrEY4jcfgbrlDB7onpgqhlDciyToEBCYK23G0HgOZkS4WAngQxB4CwFHTFQBjAAoSGYGhGGdthsgiGI7AkvwgiKCo6jcbokIGEYICmOY%2Bh6RekDTKgcS2AIIm8Kg8nEKZWARVAzBsCASo5AwpAwWIJiLPSjKzpYWmMSkDgMM4rj1Ho/ijAURR6AkSSxaktUZJCrU5R0TXdJCTTta0/SdZ4g3lc0DAjSM%2BRdFEg3DAM429G0fXzRI0zDnMCybQeHDtqQnbdr2HArKoqKIjqiKSCsADiThOH2QgrBAD1PS96zmJE2wQLghAkOO0RcJMvDXlokxziAYb7UepDmZIGJBmuiKMuS0SIhoa4Ysix0JWeliXlOM7TPeT6zAQcQ3J%2BEDfm%2B9DEMErCLCFBDIMGq5Bt22VECleiecIojiB5sjeWoJ7%2BaQyHEEwcQWfth146eHCoTcVNXKgsYXVdN33Y9z2ve9BtfZpxC/S4P4M0DINgyTLaHrw5lhkGUjRF4W4Y9EGJSBo0hK6d55E%2BDpXzvyLvRJIkhEmGkf0l4WLadEx7cQHxM3qTj5kxASB07%2BNO5wzwVSBiuKwfBmB2Q5mGiXwdCm/hhHcVRZEUaQzc0XRDE2K3LGMAQ7GcSevH8YJtDCa34mGMAUndjJFXySJ3PKcgqmtxpWknrp%2Bk0YZizdiZZny1ZTA2ZX6HV85giucL0gC2LvndroMKBcYg5hWEaVRTFKTxT2SUpQpSKZVrDtSqjVdIK0GpzXGAtLIbUUjLRatkdq60YF6CGhUJaY10GTWGsMVBzVFptEQUQmoBCBpbRHLtEGCtk4nU4CsVm7MuBLiRg6P6%2BBebW1BmnCG0wECYCYFgKIWxSDzhxLibScMna4n9gTC8V4SZ3kzkgCm6t84vnppEJmmUODa2urdI2L03r60%2Bkwb6ZteA8xIHgdA/MXJC3crfUWShxZ%2BQXFLGWctRL2wOnQ/GKs1Y3BWJrRhg5mGsK5q9C2WigSWj3Dw4OkNSACKEd0UR4idyw0dtDWRJ5U4KN4SHCQUIgxeFBJyKQoI9xuxXMufaScjr5IJkk5R8As4500XnCgtMumF2IJSLJZdEIoXPk5XgOF66UEbt2duXFJxzM7oxHuL5WL9w4lxWemA%2BICSEiJSck9JJ7zEngWStgF6KWXqvGu69tLdi3jLHeGBjlTlMuZHxfBrK2VGY5VuAtr5OJkF5VxD8dA9BfsFN%2BW9P49m/nFAmiVIgAJhRg%2BwEBHAkN8NVchsCertUxXilIOKcEgMwcQ7BE1SUtHwY1DapDagUtWmQ2laCaHbVHHtbSitmlnSYcGLwGgokcIBnEiciS7YpMEcIygUN2ShlBJIL2GhQSImiCqyQqIMToikTkmGciOCB0UenNpsBVFBIsD0gu2jmacD5SwxGXMrGcJsXYyE/zHESGccCnyEsejS1lvLLl/jlaq0psE0JdqymCvYTE38QMpi23TvwqV6TfHSNyU0lO8ig4SvnMDIMqrUZElRBHSQGgCQBU4I0/Via%2BH7ShMG1OSTphJSSPYSQQA%3D%3D%3D)). But keeping the orginal message, will fix this issue

Reply:
Yeah, because we can't compile for loops properly yet. I agree that keeping the original error message is better for now :)